Tests/TocTests.fs

```diff
@@ -75,13 +148,195 @@ module RenderToc =
         let expectedLines =
             [| StartMarker
                "- [T1](#t1)"
-               " - [T2](#t2)"
-               "  - [T3](#t3)"
-               " - [T4](#t4)"
-               "  - [T5](#t5)"
-               EndMarker
-               "" |]
+               "  - [T2](#t2)"
+               "    - [T3](#t3)"
+               "  - [T4](#t4)"
+               "    - [T5](#t5)"
+               Toc.EndMarker |]
 
         let expected = String.concat NewLine expectedLines
 
         Assert.Equal(expected, titles)
+
+module DocumentEdit =
+    [<Fact>]
+    let insert_afterYaml () =
+        let text = 
+            stripMarginTrim
+                $"
+                |---
+                |title: \"First\"
+                |tags: [1, 2]
+                |---
+                |
+                |## T1 
+                |### T2
+                |## T3
+                |#### T4
+                "
+        let doc = FakeDoc.Mk text
+
+        let action = CodeActions.tableOfContents doc |> Option.get
+
+        let modifiedText = applyDocumentAction doc action
+
+        let expected =
```

Comment:
nit: I'm not sure whether it'll be better in this case, but you could try snapshot tests here to avoid manually typing all these expected doc contents.